performance/buildSrc/src/main/kotlin/RegressionsReporter.kt

```diff
@@ -0,0 +1,252 @@
+ */
+
+import groovy.lang.Closure
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+
+import com.ullink.slack.simpleslackapi.impl.SlackSessionFactory
+
+import org.jetbrains.report.json.*
+
+import java.io.FileInputStream
+import java.io.IOException
+import java.io.File
+import java.util.concurrent.TimeUnit
+import java.net.HttpURLConnection
+import java.net.URL
+import java.util.Base64
+import java.util.Properties
+
+// Run command line from string.
+fun String.runCommand(workingDir: File = File("."),
+                      timeoutAmount: Long = 60,
+                      timeoutUnit: TimeUnit = TimeUnit.SECONDS): String {
+    return try {
+        ProcessBuilder(*this.split("\\s".toRegex()).toTypedArray())
+                .directory(workingDir)
+                .redirectOutput(ProcessBuilder.Redirect.PIPE)
+                .redirectError(ProcessBuilder.Redirect.PIPE)
+                .start().apply {
+                    waitFor(timeoutAmount, timeoutUnit)
+                }.inputStream.bufferedReader().readText()
+    } catch (e: IOException) {
+        error("Couldn't run command $this")
+    }
+}
+
+// List of commits.
+class CommitsList(data: JsonElement): ConvertedFromJson {
+
+    val commits: List<Triple<String, String, String>>
+
+    init {
+        if (data is JsonObject) {
+            val changesElement = getRequiredField(data, "change")
+            if (changesElement is JsonArray) {
+                commits = arrayToList(changesElement.jsonArray, { index ->
+                    if (getObjectOrNull(index) != null)
+                        Triple(elementToString(getRequiredField((this.getObjectOrNull(index) as JsonObject), "version"), "version"),
+                                elementToString(getRequiredField((this.getObjectOrNull(index) as JsonObject), "username"), "username"),
+                                elementToString(getRequiredField((this.getObjectOrNull(index) as JsonObject), "webUrl"), "webUrl")
+                        )
+                    else null
+                })
+            } else {
+                error("Change field is expected to be an array. Please, check source.")
+            }
+        } else {
+            error("Commits description is expected to be a json object!")
+        }
+    }
+}
+
+/**
+ * Task to produce regressions report and send it to slack. Requires a report with current benchmarks result
+ * and path to analyzer tool
+ *
+ * @property currentBenchmarksReportFile  path to file with becnhmarks result
+ * @property analyzer path to analyzer tool
+ * @property fileNameForPreviousResults name of file where should be saved benchmarks results from previous build
+ */
+open class RegressionsReporter : DefaultTask() {
+
+    val teamCityUrl = "http://buildserver.labs.intellij.net"
+    val defaultBranch = "master"
```

Comment:
The default branch may differ in different TeamCity projects (e.g. the Big Kotlin has `dev` TC-project where `master` is a default branch and `1.3.20` TC-project where is the default branch is `1.3.20`). So it may make sense to read the default branch from build/project properties. 

I don't think we need to do it right now, though. Just a thought for the future :)

Reply:
Let's postpone this, default branch in TC can be got via one more rest request or from gradle, but we planned migrating of saved artifacts to bintray, so it'll be reworked later.